### ISSUE TITLE
No state change errors after upgrading from 0.0.3 to 0.2.1

### ISSUE THREAD

**Issue by sebakerckhof** [Urigo/angular-meteor-auth#7]
Hello,

In my code I catch state change errors from angular-meteor-auth to redirect users to the login page.
Here's some simplified code:
```js
angular.module('app',[...]).run(redirectLogin).config(routes);

redirectLogin.$inject =  ["$rootScope", "$state"];
function redirectLogin($rootScope, $state){
  $rootScope.$on("$stateChangeError", function(event, toState, toParams, fromState, fromParams, error) {
    if (error === "AUTH_REQUIRED") {
      $state.go('auth.login');
    }
  });
}

routes.$inject = ['$stateProvider', '$locationProvider'];
function routes($stateProvider, $locationProvider){
    $locationProvider.html5Mode(true);

    $stateProvider
      .state('default', {
        url: '/',
        templateUrl: '/packages/barco:meeting-frontend/client/landing/view/landing.html',
        controller: 'LandingCtrl',
        controllerAs: 'vm',
        resolve: {
          "currentUser": ["$auth", function($auth) {
            return $auth.requireUser();
          }]
        }
      });
}
```

This worked using angular-meteor `1.3.4` and angular-meteor-auth `0.0.2`. 
I upgraded to the latest (`1.3.7` and `0.2.1`) and I never get into the `$stateChangeError` callback. I do however get to this line: https://github.com/Urigo/angular-meteor-auth/blob/master/dist/angular-meteor-auth.bundle.js#L94, so somewhere in between something must have changed which breaks my code, but what would it be?

Btw, the code in this repo seems not to be the latest? (it's `0.2.0-beta` instead of `0.2.1`) and the code seems to be quite different from what I get from atmosphere....

**Comment by sebakerckhof** [comment]
Very weird, but if I replace:
```js
 this.requireUser = (c) => {
     let waiting = this.waitForUser();
   
     let promise = waiting.then((currentUser) => {
       if (currentUser)
         return $q.resolve(currentUser);
   
       return $q.reject('AUTH_REQUIRED');
     });
   
     promise.stop = waiting.stop;
     return promise;
};
```
With:
```js
this.requireUser = (c) => {
   let waiting = this.waitForUser();

   let promise = waiting.then((currentUser) => new Promise((resolve,reject) => {
     if (currentUser) resolve(currentUser);
     else reject('AUTH_REQUIRED');
   }));

   promise.stop = waiting.stop;
   return promise;
 };
```

Everything works again, although they should do exactly the same thing...

**Comment by Urigo** [comment]
Hi @sebakerckhof , haven't had time to look at this yet, but I've just released a new major version (`1.0.0`).
Can you try testing it with angular-meteor `1.3.9` and see if that still happens?
If so I will start working on it soon.
thanks

**Comment by sebakerckhof** [comment]
I still have problems with the newest version, but I'll make a new issue for that one.
